src/presets/aws-lambda-edge.ts

```diff
@@ -0,0 +1,167 @@
+import { resolve } from "pathe";
+import { defineNitroPreset } from "../preset";
+import { Nitro } from "../types";
+import { writeFile } from "../utils";
+
+export const awsLambdaEdge = defineNitroPreset({
+  entry: "#internal/nitro/entries/aws-lambda-edge",
+  externals: true,
+  commands: {
+    deploy: "cd ./cdk && APP_ID=<your app id> npm run deploy --all",
+  },
+  hooks: {
+    async compiled(nitro: Nitro) {
+      await generateCdkApp(nitro);
+    },
+  },
+});
+
+async function generateCdkApp(nitro: Nitro) {
+  const cdkDir = resolve(nitro.options.output.dir, "cdk");
+  await writeFile(resolve(cdkDir, "bin/nitro-lambda-edge.ts"), entryTemplate());
+  await writeFile(
+    resolve(cdkDir, "lib/nitro-lambda-edge-stack.ts"),
+    nitroLambdaEdgeStackTemplate()
+  );
+  await writeFile(
+    resolve(cdkDir, "package.json"),
+    JSON.stringify({
+      private: true,
+      scripts: {
+        cdk: "cdk",
+        deploy: "npm install && cdk deploy",
+      },
+      devDependencies: {
+        "@types/node": "18",
+        "aws-cdk": "^2",
+        "ts-node": "^10.7.0",
+        typescript: "~3.9.7",
+      },
+      dependencies: {
+        "aws-cdk-lib": "^2",
+        constructs: "^10.0.0",
+        "nitro-aws-cdk-lib": "latest",
+        "source-map-support": "^0.5.21",
+      },
+    })
+  );
+  await writeFile(
+    resolve(cdkDir, "cdk.json"),
+    JSON.stringify({
+      app: "npx ts-node --prefer-ts-exts bin/nitro-lambda-edge.ts",
```

Comment:
could use jiti instead of ts-node here

Reply:
I didn't know jiti before I had recieve this comment! Thank you!!